ggml.c

```diff
@@ -3689,6 +3691,8 @@ struct ggml_context * ggml_init(struct ggml_init_params params) {
         // initialize cuBLAS
         #if defined(GGML_USE_CUBLAS)
         ggml_init_cublas();
+        #elif defined(GGML_USE_CLBLAST)
+        ggml_cl_init();
```

Comment:
I don't think this is a good practice; I would prefer if the cl-compiled code can detect if there are any devices; select 1 (and warn if there are multiple), and if there are no devices, fall back on the CPU implementation.

Reply:
That's not how the othe BLAS's work right now, if they are enabled they are built in and can't be disabled. Not the best system because there is actually a check for every matmul already that could disable them.

Reply:
I think this can be a future improvement.